src/pages/index.js

```diff
@@ -34,50 +36,126 @@ const enableValidation = (formSettings) => {
 
 enableValidation(formSettings);
 
+/* ************************* */
+/* Declaration of Api object */
+/* ************************* */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authorization: "a0548a22-3cdf-48eb-9822-7e7daf1b0604",
+});
+
 /* ****************************** */
 /* Declaration of UserInfo object */
 /* ****************************** */
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__job",
+  avatarSelector: ".profile__picture",
 });
 
 /* ********************** */
 /* Declarations of popups */
 /* ********************** */
+const avatarForm = new PopupWithForm("form[name='avatar-form']", (data) => {
+  avatarForm.submitButton.textContent = "Saving...";
+  api
+    .editProfilePicture(data.link)
+    .then((res) => (res.ok ? true : res.status))
+    .then(() => {
+      userInfo.setAvatar(data.link);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      avatarForm.submitButton.textContent = "Save";
+      avatarForm.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


**Please, correct it everywhere**